pkg/resourcevalidator/info.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcevalidator
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+type peeringInfo struct {
+	clusterID        string
+	clusterName      string
+	SPList           map[string]ShadowPodDescription
+	PeeringQuota     v1.ResourceList
+	FreePeeringQuota v1.ResourceList
```

Comment:
maybe i didn't understand, sorry

Reply:
Here, you are storing `PeeringQuota` (i.e., the total amount of resources offered), and `FreePeeringQuota` (i.e., total - used). I would personally feel easier to consider `PeeringQuota` and `UsedPeeringQuota` (i.e., used), as the latter is unrelated from the total amount of resources offered. Otherwise, in case the total changes, also the free amount needs to be updated. Just personal feeling though.

Reply:
Yeah, now i understand. This choice was made to speed up the resource check of each ShadowPod. in this way if `ShadowPod resources` fit the `FreePeeringQuota` it is immediately accepted. Even if in the way you suggest i can also firstly do a subtraction to obtain the `FreePeeringQuota`. Tell me which strategy do you prefer, for me is the same and change does not produce a particular effort

Reply:
I think that comparing against 0, or a value (i.e., the total) should introduce no different overhead. Yet, the latter approach seems less prone to possible drifts, at least to me

Reply:
ok